common/display/displayqueue.cpp

```diff
@@ -198,8 +192,7 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers) {
     overlay_layer.SetIndex(layer_index);
     overlay_layer.SetAcquireFence(layer->acquire_fence.Release());
     queue_item.layers_rects_.emplace_back(layer->GetDisplayFrame());
-    int ret = layer->release_fence.Reset(
-        queue_item.sync_object_->CreateNextTimelineFence());
+    int ret = layer->release_fence.Reset(overlay_layer.GetReleaseFence());
```

Comment:
These are fences associated with current set of layers. When these set of layers are on screen, we signal the fences of previous set (layers can be different etc doesn't matter any more since we track it for individual layers now)and tied to life time of that layer.